### ISSUE TITLE
CI: Test against nightly versions of Scala 

### ISSUE THREAD

**Issue by WojciechMazur** [scala-native/scala-native#3731]
As an ongoing effort to improve Scala release procedure it was decided at Scala Core team meeting we should detect incompatibilities with Scala compiler/runtime as early as possible. It should speed up releasing of them compiler as we would not need to wait until next RC version. 

We should extend scheduled CI to enable testing against nightly versions of Scala. It should include: 
- [ ] Test against latest Scala 3 LTS nightly version (when LTS nightlies would be enablefd)
- [ ] Test against latest Scala 3 Next 
- [ ] Test against Scala 2.13 nightly version
- [ ] Test against Scala 2.12 nightly version

TODO: 
- [ ] - Automatic detection of nightly versions: 
  For Scala 3 Next we can use simple script parsing Maven metadata https://github.com/VirtusLab/community-build3/blob/master/scripts/lastVersionNightly.sc 
  For Scala 3 LTS and Scala 2 we would need to figure out how to reliably resolve nightly versions (we assume these does not have generic name, like x.y.z-SNAPSHOT, but rather use a commit SHA as part of version)

**Comment by ldeluigi** [comment]
You could setup a [webhook triggered workflow](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#repository_dispatch) in this repo that gets called from a workflow put in the dotty/scala repositories that puts the information on nightly build, sha and so on inside the payload. You can then access that payload from the workflow as described in the gh docs.

**Comment by LeeTibbert** [comment]
@WojciechMazur @kitbellew 

Housekeeping: 

I believe this can be closed thanks to the recent work by you and kitbellew.

Progress is slow, but it feels good when it happens. Thank to both of you.

**Comment by WojciechMazur** [comment]
@LeeTibbert no, that's something different. Although, we at least test with some, fixed, Scala 3 nightly version. In fact I forgot about this topic, thank you for the reminder.
